### PR TITLE
feat: 0x1B - SHL opcode 

### PR DESCRIPTION
**Pull request by trbutler4**
implemented the shift left operator 
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
shift left operator not implemented (just returns Ok())

Resolves: #15 

## What is the new behavior?

shift left is implemented
-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/evm/src/instructions/comparison_operations.cairo [hunk 1]
```diff
@@ -110,6 +111,17 @@ impl ComparisonAndBitwiseOperations of ComparisonAndBitwiseOperationsTrait {
...
+        let shift = *popped[0];
+        let val = *popped[1];
+
+        // if shift is bigger than 255 return 0
+        if shift > 255 {
+            return self.stack.push(0);
+        }
+
+        let (shifted, _) = u256_overflow_mul(val, 2.pow_mod(shift));
+        self.stack.push(shifted)?;
```
Comment (enitrat) [inline crates/evm/src/instructions/comparison_operations.cairo:124]:
```suggestion
        self.stack.push(shifted)
```

you can directly do this instead. and ditch the next line

File: crates/evm/src/tests/test_instructions/test_comparison_operations.cairo [hunk 2]
```diff
@@ -196,3 +196,20 @@ fn test_byte_offset_out_of_range() {
     assert(ctx.stack.len() == 1, 'stack should have one element');
     assert(ctx.stack.peek().unwrap() == 0x00, 'stack top should be 0x00');
 }
+
+
+#[test]
+#[available_gas(20000000)]
+fn test_exec_shl() {
```
Comment (enitrat) [inline crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:203]: can you seperate this into two tests with explicit naming?

Comment (trbutler4) [reply crates/evm/src/tests/test_instructions/test_comparison_operations.cairo:203]: sure thing, once again after i submit

**Comment by enitrat** [comment]
@trbutler4 you can now use our bitshift library in utils::math

**Comment by trbutler4** [comment]
push is failing due to:

Running gas snapshot comparison...
.git/hooks/pre-push: 5: python: not found
Error: Snapshot comparison failed!
error: failed to push some refs to 'github.com:trbutler4/kakarot-ssj.git'

**Comment by trbutler4** [comment]
i do have python installed, and can run the python command

**Comment by enitrat** [comment]
You can run git push with the --no-verify flag to skip the hook

**Pull request closed without merging**